asteroid/data/dampvsep_dataset.py

```diff
@@ -0,0 +1,231 @@
+        self.tracks = self.get_tracks()
+        self.perf_key = [*self.tracks]  # list of performances keys
+
+        self.norm = norm
+
+    def __len__(self):
+        return len(self.tracks) * self.samples_per_track
+
+    def __getitem__(self, index):
+        audio_sources = {}
+        # get track_id
+
+        track_id = index // self.samples_per_track
+        perf = self.perf_key[track_id]
+        self.mixture_path = perf
+
+        min_duration = int(min(self.tracks[perf]['vocal_duration'],
+                               self.tracks[perf]['background_duration']))
+        # Set start time of segment
+        start = 0
+        if self.random_segments:
+            start = random.randint(
+                0, int(min_duration - self.segment))
+
+        # Set end time of segment
+        stop = min_duration
+        if self.segment:
+            stop = start + self.segment
+
+        for source in ['vocal', 'background']:
+            x, _ = librosa.load(
+                self.root_path / self.tracks[perf][source],
+                sr=self.sample_rate,
+                mono=True,
+                offset=start,
+                duration=stop - start,
+                res_type='polyphase'
+            )
+            audio_sources[source] = x
+        
+        audio_sources['vocal'] *= float(self.tracks[perf]['scaler'])
+            
+        if self.norm == 'song_level':
+            mix_mean = float(self.tracks[perf]['mean'])
+            mix_std = float(self.tracks[perf]['std'])
+        else:
+            mix_mean = 0.
+            mix_std = 1.
+
+        if self.target:
+            audio_sources = torch.stack([
+                (torch.from_numpy(wav.T) - mix_mean) / mix_std for src, wav in audio_sources.items()
+                if src in self.target
+            ], dim=0)
+        
+        audio_mix = audio_sources.sum(0)
+        audio_mix = (audio_mix - mix_mean) / mix_std
+    
+        return audio_mix, audio_sources
+    
+    def get_track_name(self, idx):
+        track_id = idx // self.samples_per_track
+        return self.perf_key[track_id]
+
+    def get_tracks(self):
+        """
+        Loads tracks
+        """
+        metadata_path = Path(f'metadata/{self.split}_sr{self.sample_rate}.json')
+        if metadata_path.exists():
+            print(f"Metadata for {self.split} set exist!! Using it.")
+            tracks = json.load(open(metadata_path, 'r'))
+        else:
+            print(f"Constructing Metadata for {self.split} set")
+            track_list = pd.read_csv(f"local/{self.split}.csv")
+            tracks = self.build_metadata(track_list)
+
+            metadata_path.parent.mkdir(parents=True, exist_ok=True)
+            json.dump(tracks, open(metadata_path, 'w'), indent=2)
+
+        return tracks
+        
+    @staticmethod
+    def _build_metadata(inputs):
+        sample, root, sample_rate = inputs
+        back, _ = librosa.load(root / sample['background_path'],
```

Comment:
We usually use `soundfile` for loading but it doesn't resample. 
Is resampling needed? If yes, isn't is expensive to resample on the fly when it could be fixed?

Reply:
Librosa it is also used in AVSpeech and they have it as a requirement in the recipe.
I could do the same